builtin/credential/okta/backend.go

```diff
@@ -168,22 +175,39 @@ func (b *backend) Login(ctx context.Context, req *logical.Request, username stri
 		factorAvailable := false
 
 		var selectedFactor mfaFactor
-		// only okta push is currently supported
+
+		// Okta push and totp are currently supported. If a totp passcode is provided during
+		// login and is supported, that will be the preferred method.
 		for _, v := range result.Embedded.Factors {
-			if v.Type == "push" && v.Provider == "OKTA" {
+			if v.Provider != "OKTA" {
+				continue
+			}
+
+			if v.Type == mfaTOTPMethod && totp != "" {
```

Comment:
Should we just check `v.Type == mfaTOTPMethod` in here and do the `totp` check separately afterwards? The error response below is not entirely accurate if a TOTP factor was set but a totp was not provided in the login request.

Reply:
That doesn't quite work since if push is available and they didn't provide `totp`, we should just use push. But I did just redo the handling of these (https://github.com/hashicorp/vault/pull/10942/commits/0514d1378d732a156172580ccb05e08c28679675) to allow separate, more specific error messages.